broker/broker.py

```diff
@@ -92,9 +92,14 @@ def __init__(self, **kwargs):
             self.host_classes.update(kwargs.pop("host_classes"))
         # determine the provider actions based on kwarg parameters
         self._provider_actions = {}
+        no_provider = False
         for key, action in PROVIDER_ACTIONS.items():
             if key in kwargs:
                 self._provider_actions[key] = action
+            else:
+                no_provider = True
+        if no_provider:
+            logger.error("Could not determine an appropriate provider")
```

Comment:
We can simplify further!
```suggestion
        for key, action in PROVIDER_ACTIONS.items():
            if key in kwargs:
                self._provider_actions[key] = action
        if not self._provider_actions:
            logger.error("Could not determine an appropriate provider")
```

Reply:
Oh I see. I didn't think to check that `_provider_actions` was null that way.